src/api/app/models/configuration.rb

```diff
@@ -71,6 +71,19 @@ def ldapgroup_enabled?
   end
   # End of class methods
 
+  def ldap_enabled?
+    CONFIG['ldap_mode'] == :on
+  end
+
+  def passwords_changable?
+    change_password && CONFIG['proxy_auth_mode'] != :on && CONFIG['ldap_mode'] != :on
+  end
+
+  def accounts_editable?
+    (CONFIG['proxy_auth_mode'] != :on || (CONFIG['proxy_auth_mode'] == :on && !CONFIG['proxy_auth_account_page'].blank?)) && \
+      CONFIG['ldap_mode'] != :on
+  end
+
```

Comment:
I totally agree with you and I first had it like you said it, but since a user can also set instead of `:off` something like `:shit` it would enable LDAP then or at least pass the condition.